kodilla-stream/src/main/java/com/kodilla/optional/homework/Student.java

```diff
@@ -17,7 +18,7 @@ public String getName() {
     }
 
     public Teacher getTeacher() {
-        return teacher;
+        return Optional.ofNullable(teacher).orElse(new Teacher("<undefined>"));
```

Comment:
masz teraz dwa razy uzyte "undefined".
Proponuje tu zrobic:
public Optional<Teacher> getTeacher() { return Optional.ofNullable(teacher); }

a w klasie Application: student.getTeacher().map(Teacher::getName).orElse("<undefined>");

Reply:
public Student(String name) {
        this.name = name;	       
        this.teacher = null;
    }